vllm/model_executor/models/minicpmv.py

```diff
@@ -568,110 +509,60 @@ def get_image_bounds(self, input_ids: torch.Tensor):
-
-    def get_embedding(self, data: Dict[str, Union[List[torch.Tensor],
-                                                  torch.Tensor]]):
-        input_ids = data["input_ids"]
-
-        vision_hidden_states = self.get_vision_hidden_states(data)
-        if vision_hidden_states is not None and len(vision_hidden_states) > 0:
+            vision_hidden_states = self.get_vision_hidden_states(image_inputs)
             image_bounds = self.get_image_bounds(input_ids)
-        else:
-            image_bounds = []
 
-        if hasattr(self.config, 'scale_emb'):
-            vlm_embedding = self.llm.embed_tokens(
-                input_ids) * self.config.scale_emb
-        else:
-            vlm_embedding = self.llm.embed_tokens(input_ids)
-        vision_hidden_states = [
-            i.type(vlm_embedding.dtype) if isinstance(i, torch.Tensor) else i
-            for i in vision_hidden_states
-        ]
+            if hasattr(self.config, "scale_emb"):
+                vlm_embedding: torch.Tensor = (
+                    self.llm.embed_tokens(input_ids) * self.config.scale_emb)
+            else:
+                vlm_embedding = self.llm.embed_tokens(input_ids)
 
-        if len(vision_hidden_states) > 0 and len(image_bounds) > 0:
-            vision_hidden_states = torch.cat(vision_hidden_states, dim=0)
             image_indices = torch.stack([
                 torch.arange(r[0], r[1], dtype=torch.long)
                 for r in image_bounds
-            ]).to(vlm_embedding.device)
+            ]).to(vlm_embedding.device, vlm_embedding.dtype)
             vlm_embedding.scatter_(
                 0,
                 image_indices.view(-1, 1).repeat(1, vlm_embedding.shape[-1]),
-                vision_hidden_states.view(-1, vision_hidden_states.shape[-1]))
+                vision_hidden_states.view(-1, vision_hidden_states.shape[-1]),
+            )
+
         return vlm_embedding, vision_hidden_states
 
-    def process_multimodal_inputs(self, inputs: Dict[str,
-                                                     Union[List[torch.Tensor],
-                                                           torch.Tensor]]):
-        pixel_values = []
-        tgt_sizes = []
-        for b in range(len(inputs["pixel_values"])):
-            pixel_values += inputs["pixel_values"][b]
-            tgt_sizes += inputs["tgt_sizes"][b]
-        return {
-            "pixel_values": pixel_values,
-            "input_ids": inputs["input_ids"],
-            "tgt_sizes": tgt_sizes
-        }
+    def _parse_and_validate_image_inputs(
+            self, **kwargs: object) -> Optional[MiniCPMVImageInputs]:
+        pixel_values = kwargs.pop("pixel_values", [])
+        tgt_sizes = kwargs.pop("tgt_sizes", [])
+
+        if not isinstance(pixel_values, (torch.Tensor, list)):
+            raise ValueError("Incorrect type of pixel values. "
+                             f"Got type: {type(pixel_values)}")
+
+        if not isinstance(tgt_sizes, torch.Tensor):
```

Comment:
And here, `tgt_sizes` should be `List` rather than `Tensor` in practice.
And modified to `(torch.Tensor, list)`

Reply:
It's not converted automatically to a tensor by the processor?

Reply:
Fixed.